src/components/TodoList/TodoList.tsx

```diff
@@ -1,224 +1,150 @@
-/* eslint-disable */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  fetchTodosStart,
+  fetchTodosSuccess,
+  fetchTodosFailure,
+} from '../../features/todos';
+import { getTodos, getUser } from '../../api';
+import { RootState } from '../../app/store';
+import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { setCurrentTodo } from '../../features/currentTodo';
+import { TodoModal } from '../TodoModal';
 
 export const TodoList: React.FC = () => {
+  const dispatch = useDispatch();
+  const { todos, loading, error } = useSelector(
+    (state: RootState) => state.todos,
+  );
+  const { status, query } = useSelector((state: RootState) => state.filter);
+  const currentTodo = useSelector((state: RootState) => state.currentTodo);
+  const [isModalOpen, setModalOpen] = useState(false);
+  const [currentUser, setCurrentUser] = useState<User | null>(null);
+  const [selectedTodoId, setSelectedTodoId] = useState<number | null>(null);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      dispatch(fetchTodosStart());
+      try {
+        const todosData = await getTodos();
+
+        dispatch(fetchTodosSuccess(todosData));
+      } catch (fetchError) {
+        dispatch(fetchTodosFailure('Error fetching todos'));
+      }
+    };
+
+    fetchData();
+  }, [dispatch]);
+
+  const filteredTodos = todos.filter((todo: Todo) => {
+    if (status === 'active' && todo.completed) {
+      return false;
+    }
+
+    if (status === 'completed' && !todo.completed) {
+      return false;
+    }
+
+    if (!todo.title.toLowerCase().includes(query.toLowerCase())) {
+      return false;
+    }
+
+    return true;
+  });
+
+  const handleSelectTodo = async (todo: Todo) => {
+    if (selectedTodoId === todo.id) {
+      setModalOpen(!isModalOpen);
+      setSelectedTodoId(selectedTodoId ? null : todo.id);
+    } else {
+      dispatch(setCurrentTodo(todo));
+      const userData = await getUser(todo.userId);
+
+      setCurrentUser(userData);
+      setSelectedTodoId(todo.id);
+      setModalOpen(true);
+    }
+  };
+
+  const handleCloseModal = () => {
+    setModalOpen(false);
+    setCurrentUser(null);
+    setSelectedTodoId(null);
+  };
```

Comment:
This function resets the selected user and todo when the modal is closed. However, this could lead to unnecessary re-renders when the modal is re-opened with the same todo. Consider keeping the selected user and todo in the state until a different todo is selected.